### ISSUE TITLE
Live chat widget 2.0 fails when customization file is bigger than x

### ISSUE THREAD

**Issue by ajj10** [microsoft/omnichannel-chat-widget#283]
**Describe the bug**
So I am trying to utilize the new live chat widget 2.0 with the v2 script and a "data-customization-callback" data tag with some customizations to go with it. As soon as The customization file gets bigger than x long the chat fails to load.

**To Reproduce**
1. Set up any basic website and connect to a omnichannel envoirement with valid data-app-id, data-org-id and data-org-url:
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Eimskip DEV Chat</title>
</head>

<body style="margin: 0">
    <script src="lcwCustomization1.js"></script>

    
    <script v2 id="Microsoft_Omnichannel_LCWidget"
        src="https://oc-cdn-public-eur.azureedge.net/livechatwidget/scripts/LiveChatBootstrapper.js"
        data-app-id="xxxxxxxx-xxxxx-xxxx-xxxxx" data-font-family-override="Inter, Arial, sans-serif"
        data-org-id="xxxxxxxx-xxxxx-xxxx-xxxxx" data-lcw-version="prod" data-color-override="#dc2919"
        data-org-url="xxxxxxxx-xxxxx-xxxx-xxxxx"
        data-customization-callback="lcwCustomization">
        </script>

</body>

</html>

2. include the customization file lcwCustomization1.js:
function lcwCustomization() {
    return {
        preChatSurveyPaneProps: {
            controlProps: {
                payload: "{\"type\":\"AdaptiveCard\",\"version\":\"1.1\",\"body\":[{\"type\":\"TextBlock\",\"text\":\" Nafn\"},{\"type\":\"Input.Text\",\"id\":\"{\\\"Id\\\":\\\"ed2492e1-daa7-ed11-aad0-0022489f8e97\\\",\\\"Name\\\":\\\"TestSpurning\\\",\\\"IsOption\\\":false,\\\"Order\\\":1,\\\"IsRequired\\\":true,\\\"QuestionText\\\":\\\"Nafn\\\"}\",\"isRequired\":true,\"errorMessage\":\"Vinsamlegast fyllið út\"},{\"type\":\"TextBlock\",\"text\":\" Netfang\"},{\"type\":\"Input.Text\",\"id\":\"{\\\"Id\\\":\\\"8137ea68-09b1-ed11-83ff-000d3aba3424\\\",\\\"Name\\\":\\\"Netfang\\\",\\\"IsOption\\\":false,\\\"Order\\\":2,\\\"IsRequired\\\":true,\\\"QuestionText\\\":\\\"Netfang\\\"}\",\"isRequired\":true,\"errorMessage\":\"Vinsamlegast fyllið út\"},{\"type\":\"TextBlock\",\"text\":\"Hvernig getum við aðstoðað?\"},{\"type\":\"Input.ChoiceSet\",\"id\":\"{\\\"Id\\\":\\\"1945d739-f4a7-ed11-aad0-0022489f8e97\\\",\\\"Name\\\":\\\"HelpType\\\",\\\"IsOption\\\":true,\\\"Order\\\":3,\\\"IsRequired\\\":true,\\\"QuestionText\\\":\\\"Hvernig\\\"}\",\"value\":\"{\\\"Id\\\":\\\"8cd\\\",\\\"Value\\\":\\\"-\\\"}\",\"isMultiSelect\":false,\"choices\":[{\"title\":\"-\",\"value\":\"{\\\"Id\\\":\\\"8cdf5103-e1bc-ed11-83ff-000d3aba3cb5\\\",\\\"Value\\\":\\\"-\\\"}\"},{\"title\":\"Innanlandsflutningar\",\"value\":\"{\\\"Id\\\":\\\"8ddf5103-e1bc-ed11-83ff-000d3aba3cb5\\\",\\\"Value\\\":\\\"Innanlandsflutningar\\\"}\"},{\"title\":\"Millilandaflutningar\",\"value\":\"{\\\"Id\\\":\\\"8bdf5103-e1bc-ed11-83ff-000d3aba3cb5\\\",\\\"Value\\\":\\\"Millilandaflutningar\\\"}\"},{\"title\":\"Tollskjalagerð\",\"value\":\"{\\\"Id\\\":\\\"8fdf5103-e1bc-ed11-83ff-000d3aba3cb5\\\",\\\"Value\\\":\\\"Tollskjalagerð\\\"}\"}],\"isRequired\":true},{\"type\":\"Input.Toggle\",\"id\":\"{\\\"Id\\\":\\\"2444f10b-12e4-ed11-a7c7-000d3aba3cb5\\\",\\\"Name\\\":\\\"Agreement\\\",\\\"IsOption\\\":false,\\\"Order\\\":4,\\\"IsRequired\\\":false,\\\"QuestionText\\\":\\\"Samþykki\\\"}\",\"title\":\"Samþykki Persónuverndarstefnu Eimskips sem finna má *[hér](https://www.eimskip.com/privacy-policy/)*. Vinsamlegast deilið ekki viðkvæmum persónuupplýsingum í netspjalli.\",\"valueOn\":\"true\",\"valueOff\":\"false\",\"value\":\"false\",\"wrap\":true,\"isRequired\":true,\"errorMessage\":\"Vinsamlegast fyllið út\"}],\"actions\":[{\"type\":\"Action.Submit\",\"data\":{\"Type\":\"InputSubmit\"},\"title\":\"SENDA\"}]}"
            }
        },

        chatButtonProps: {
            controlProps: {
                //hideChatTitle: true,
                hideChatTextContainer: true
            },
            styleProps: {
                generalStyleProps: {
                    height: "60px",
                    width: "60px",
                    selectors: {
                        ":focus": {
                            outline: "none"
                        }
                    }
                },
                iconStyleProps: {
                    width: "60px",
                    height: "60px",
                    borderWidth: "2px"
                }
            }
        },

        outOfOfficeChatButtonProps: {
            controlProps: {
                hideChatTextContainer: true
            },
            styleProps: {
                generalStyleProps: {
                    height: "60px",
                    width: "60px",
                    selectors: {
                        ":focus": {
                            outline: "none"
                        }
                    }
                },
                iconStyleProps: {
                    backgroundColor: "#5a5d5e",
                    width: "60px",
                    height: "60px"
                }
            }
        },

        headerProps: {
            styleProps: {
                closeButtonHoverStyleProps: {
                    backgroundColor: "#dc2919",
                    filter: "brightness(0.9)"
                },
                minimizeButtonHoverStyleProps: {
                    filter: "brightness(0.9)",
                    backgroundColor: "#dc2919"
                }
            },
            controlProps: {
                closeButtonProps: {
                    ariaLabel: "Loka"
                },
                minimizeButtonProps: {
                    ariaLabel: "Minnka"
                }
            }
        },

        outOfOfficeHeaderProps: {
            styleProps: {
                generalStyleProps: {
                    backgroundColor: "#5a5d5e"
                },
                minimizeButtonHoverStyleProps: {
                    filter: "brightness(0.9)",
                    backgroundColor: "#5a5d5e"
                }
            },
            controlProps: {
                minimizeButtonProps: {
                    ariaLabel: "Minnka"
                }
            }
        },

        webChatContainerProps: {
            webChatStyles: {
                sendBoxPlaceholderColor: "#ffffff"
            },
            localizedTexts: {
                MIDDLEWARE_TYPING_INDICATOR_ONE: "{0} er að skrifa ...",
                MIDDLEWARE_MESSAGE_SENDING: "Sendist ...",
                MIDDLEWARE_MESSAGE_DELIVERED: "Sent",
            }
        },

        loadingPaneProps: {
            controlProps: {
                titleText: "Netspjall Eimskips",
                subtitleText: "",
                spinnerText: "Hinkrið aðeins ..."
            }
        },

        outOfOfficeHoursPaneProps: {
            controlProps: {
                titleText: "Spjallið okkar er opið alla virka daga 8-16 nema föstudaga 8-15"
            }
        },

        footerProps: {
            controlProps: {
                downloadTranscriptButtonProps: {
                    ariaLabel: "Sækja afrit af spjallinu"
                }
            },
            styleProps: {
                downloadTranscriptButtonStyleProps: {
                    icon: {
                        color: "#3370B7"
                    }
                },
                downloadTranscriptButtonHoverStyleProps: {
                    filter: "none",
                    backgroundColor: "#F3F2F1"
                }
            }
        },

        confirmationPaneProps: {
            controlProps: {
                titleText: "Viltu loka spjallinu?",
                subtitleText: "Til að fá afrit veldu \"Nei\" og veldu að sækja afrit neðst í vinstra horni",
                confirmButtonText: "Já",
                confirmButtonAriaLabel: "Já",
                cancelButtonText: "Nei",
                cancelButtonAriaLabel: "Nei",
            }
        }
    }
}

3. Open the html file and see widget working fine, then try uncommenting line 11 in lcwCustomization1.js and open html file again.
4. response: failed to load resource: the server responded with a status of 404 () when loading the "src" resource of the iframe
5. When going to the src of the iframe: 

**Expected behavior**
Widget running for all valid customizations.

**Screenshots**
iframe src when error is happening:
![image](https://github.com/microsoft/omnichannel-chat-widget/assets/37146798/779955b7-e018-424e-aec0-9e9b0b272ec8)

**Comment by charliewang95** [comment]
When customization string is longer than 8000 characters the iframe url will fail (browser limitation). We have pushed a fix that is rolling out depending on the region of widget source you're' using. After the fix, you can pass in a json directly to data-customization-callback (data-customization-callback="blabla.json") instead of defining everything in the same html.

On another note, since you're using script offering, this is not the best place to report issues, as this repo is focusing specifically on the <LiveChatWidget> React component. Please raise formal support tickets to Microsoft for future issues

**Comment by ajj10** [comment]
Thank you for the information! looking forward to see the changes and next time I will go through Microsoft support :)
